src/transport_portals4.h

```diff
@@ -1314,4 +1314,79 @@ void shmem_transport_syncmem(void)
     PtlAtomicSync();
 }
 
+#ifdef ENABLE_THREADS
+#define SHMEM_TRANSPORT_PTL4_CNTR_LOCK(ctx) SHMEM_MUTEX_LOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_UNLOCK(ctx) SHMEM_MUTEX_UNLOCK(ctx)
+#define SHMEM_TRANSPORT_PTL4_CNTR_READ(cntr) *(cntr) 
+#else 
+#define SHMEM_TRANSPORT_PTL4_CNTR_LOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_UNLOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_READ(cntr) shmem_internal_atomic_read(cntr) 
+#endif
```

Comment:
There are no context locks in the Portals transport (similar to FI_THREAD_SAFE build in the OFI transport).  You should just be able to `shmem_internal_atomic_read` the pending counters and `PtlCTGet` the event counters.